src/DropKit.ts

```diff
@@ -66,69 +69,88 @@ export default class DropKit {
 
       this.walletAddress = await signerOrProvider.getAddress()
       this.contract = new ethers.Contract(data.address, abi, signerOrProvider)
+      if (!this.contract) {
+        throw new Error('Initialization failed')
+      }
     }
 
     return data.address
   }
 
-  static async create(key: string, isDev?: boolean): Promise<DropKit> {
-    const dropKit = new DropKit(key, isDev)
-    await dropKit.init()
-    return dropKit
+  static async create(key: string, isDev?: boolean): Promise<DropKit | null> {
+    try {
+      const dropKit = new DropKit(key, isDev)
+      await dropKit.init()
+      return dropKit
+    } catch (error) {
+      handleError(error as EthereumRpcError<unknown>)
+      return null
+    }
   }
 
   async price(): Promise<number> {
-    if (!this.contract) {
-      throw new Error('Initialization failed')
-    }
-
     const dropPrice =
       this.version <= 3
-        ? await this.contract._price()
-        : await this.contract.price()
+        ? await this.contract?._price()
+        : await this.contract?.price()
+
     return Number(ethers.utils.formatEther(dropPrice))
   }
 
   async maxPerMint(): Promise<number> {
-    if (!this.contract) {
-      throw new Error('Initialization failed')
-    }
-
     const maxMint =
       this.version <= 3
-        ? await this.contract._maxPerMint()
-        : await this.contract.maxPerMint()
+        ? await this.contract?._maxPerMint()
+        : await this.contract?.maxPerMint()
 
     return maxMint.toNumber()
   }
 
   async maxPerWallet(): Promise<number> {
-    if (!this.contract) {
-      throw new Error('Initialization failed')
-    }
-
     const maxWallet =
       this.version <= 3
-        ? await this.contract._maxPerWallet()
-        : await this.contract.maxPerWallet()
+        ? await this.contract?._maxPerWallet()
+        : await this.contract?.maxPerWallet()
 
     return maxWallet.toNumber()
   }
 
+  async walletTokensCount(): Promise<number> {
+    return await this.contract?.balanceOf(this.walletAddress)
+  }
+
   async totalSupply(): Promise<number> {
-    if (!this.contract) {
-      throw new Error('Initialization failed')
+    const mintedNfts = await this.contract?.totalSupply()
+    return mintedNfts.toNumber()
+  }
+
+  async saleActive(): Promise<boolean> {
+    const saleActive =
+      this.version <= 3
+        ? await this.contract?.started()
+        : await this.contract?.saleActive()
+
+    return saleActive
```

Comment:
`saleActive` could be `undefined` because we're using `?`, was this throwing any typescript issues?

Reply:
Yes,  had to add `?` to all the contract functions because typescript was throwing: `Object is possibly 'null'.` even though I moved the following check into the ctor:

```typescript
if (!this.contract) {
    throw new Error('Initialization failed')
}
```

Will find a way to remove all `?` operators when accessing the contract functions

Reply:
yeah. we could use `!` instead, since `?` would return possible `null`

Reply:
Removed all `?` operators by initializing the `contract` property like so: `contract: Contract = {} as Contract`